### PR TITLE
Fix defendant uplifts on supplementary claims

### PR DESCRIPTION
**Pull request by jsugarman**
#### What
Ensure feature test behave as prod, which currently has fee calc
edge case bug, and fix that bug.

#### To duplicate bug
On staging
 - Create a scheme 11 supplementary claim
 - Add two defendants
 - Add Misc fee of "Confiscation hearing (half day)" with quantity of one - rate: £131
 - Add Misc fee of "Confiscation hearing (half day uplift)" with quantity of one - rate £52.40
 Last fee should have a rate of £26.20 (this works on local machine)

#### Ticket

[https://dsdmoj.atlassian.net/browse/CFP-105](https://dsdmoj.atlassian.net/browse/CFP-105)

#### Why
Invalid calculations for the Supplementary claim
"Confiscation hearing (half day uplift)" are causing
invalid amounts to be calculated and thereby potentially
claimed.

#### How
Suttle differences in ordering of strings with braces between local hosted postgres,
circleci alpine postgres images and AWS RDS DB instances are causing an ordering difference.
This would be fine except that a minor javascript oversight then causes a double counting of the first
fee type on the page - which happens to be the non-uplift fee quantity "Confiscation hearing (half day)".
This doubling then causes a doubling of the fee types defendant uplift "Confiscation hearing (half day uplift)"
if it is claimed/visible.

This Javascript bug has no impact if the uplift is the first fee type on the page.

Local postgres 9.6.21 on darwin puts the uplift above the non-uplift equivalent so no doubling occurs. The test
suite used alpine hosted postgres which also puts put the uplift above the non-uplift so no doubling occurs, however
when a postgres image hosted on debian is used the reverse ordering happens and the bug occurs, and the feature test
fails.

So this PR uses a debian version of postgres to match prod-like environments as closely as possible. This PR also
enforces the ordering of fee types so that postgres versions will not have an impact in any event. The result being
that non-uplift fee types should always appear above there uplift fee type.

Finally the javascript bug is fixed by removing a misplaced class, `.fx-fee-group`, on the outer container
for all the fee type checkboxes. This is causing the first fee type to be double-counted because of line
`231` of `app/webpack/javascripts/modules/external_users/claims/fee_calculator/FeeCalculator.UnitPrice.js`

### CODE REVIEW

File: app/models/fee/misc_fee_type.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ class Fee::MiscFeeType < Fee::BaseFeeType
   AGFS_SUPPLEMENTARY_TYPES = (AGFS_SUPPLEMENTARY_ONLY_TYPES + AGFS_SUPPLEMENTARY_SHARED_TYPES).freeze
   TRIAL_ONLY_TYPES = %w[MIUMU MIUMO].freeze
 
-  default_scope { order(description: :asc) }
+  default_scope { order(Arel.sql("regexp_replace(\"fee_types\".\"description\", '[()]', '', 'g')")) }
```
Comment (jrmhaig) [inline app/models/fee/misc_fee_type.rb:8]: Is the difference in ordering to do with collation? See [this Stack Overflow question.](https://stackoverflow.com/questions/56395711/postgresql-sort-a-character-field-with-special-characters-numbers-and-alphabe)

File: features/claims/advocate/scheme_eleven/advocate_supplementary_claim_submit.feature [hunk 2]
```diff
@@ -44,16 +44,16 @@ Feature: Advocate tries to submit a supplementary claim for miscellaneous fees (
       | miscellaneous | Wasted preparation fee |
 
     When I select an advocate category of 'Junior'
-    And I choose the 'Confiscation hearings (half day uplift)' miscellaneous fee with quantity of '1'
     And I choose the 'Confiscation hearings (half day)' miscellaneous fee with quantity of '2'
+    And I choose the 'Confiscation hearings (half day uplift)' miscellaneous fee with quantity of '1'
```
Comment (jrmhaig) [inline features/claims/advocate/scheme_eleven/advocate_supplementary_claim_submit.feature:48]: Does the ordering of these matter?
* If no, then do the steps need to be changed?
* If yes, then do both cases need to be tested?

File: app/models/fee/misc_fee_type.rb (hunk 1 above)
Comment (jsugarman) [reply app/models/fee/misc_fee_type.rb:8]: you'd hope so, but I tried collation first and it did not work

Comment (jsugarman) [reply app/models/fee/misc_fee_type.rb:8]: tried again but none of the following do what i'd expect/want for the unit test
```
default_scope { order(Arel.sql("\"fee_types\".\"description\" COLLATE \"POSIX\""))}
default_scope { order(Arel.sql("\"fee_types\".\"description\" COLLATE \"C\""))}
```
or any of the regional/language collations either

Comment (jsugarman) [reply app/models/fee/misc_fee_type.rb:8]: it seems to be differences between darwin/alpine and debian/amazon-linux

File: features/claims/advocate/scheme_eleven/advocate_supplementary_claim_submit.feature (hunk 2 above)
Comment (jsugarman) [reply features/claims/advocate/scheme_eleven/advocate_supplementary_claim_submit.feature:48]: no they don't, as mentioned in the commit they are merely to reflect the expected ordering

Comment (jsugarman) [reply features/claims/advocate/scheme_eleven/advocate_supplementary_claim_submit.feature:48]: yes they do need testing. this was the only test at the time that caught the problem and is a general blackbox feature test

**Review by jrmhaig — APPROVED** [review]
